xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/template/InternalTemplateManager.java

```diff
@@ -204,32 +223,55 @@ public String getPath()
-                String strinContent;
-
-                try (InputSource source = this.resource.getInputSource()) {
-                    if (source instanceof StringInputSource) {
-                        strinContent = source.toString();
-                    } else if (source instanceof ReaderInputSource) {
-                        strinContent = IOUtils.toString(((ReaderInputSource) source).getReader());
-                    } else if (source instanceof InputStreamInputSource) {
-                        // It's impossible to know the real attachment encoding, but let's assume that they respect the
-                        // standard and use UTF-8 (which is required for the files located on the filesystem)
-                        strinContent = IOUtils.toString(((InputStreamInputSource) source).getInputStream(),
-                            StandardCharsets.UTF_8);
-                    } else {
-                        return null;
-                    }
+                this.instant = this.resource.getInstant();
+                this.content = loadContent();
+            } else if (this.instant != null) {
+                // Check if the resource has been modified
+                Instant resourceInstant = this.resource.getInstant();
+                if (resourceInstant.isAfter(this.instant)) {
+                    // The resource changed, reload it
+                    this.instant = resourceInstant;
+                    this.content = loadContent();
                 }
-
-                this.content = getContentInternal(strinContent);
             }
 
             return this.content;
         }
 
+        protected T loadContent() throws Exception
+        {
+            String strinContent;
+
+            try (InputSource source = this.resource.getInputSource()) {
+                if (source instanceof StringInputSource) {
+                    strinContent = source.toString();
+                } else if (source instanceof ReaderInputSource) {
+                    strinContent = IOUtils.toString(((ReaderInputSource) source).getReader());
+                } else if (source instanceof InputStreamInputSource) {
+                    // It's impossible to know the real attachment encoding, but let's assume that they respect the
+                    // standard and use UTF-8 (which is required for the files located on the filesystem)
+                    strinContent =
+                        IOUtils.toString(((InputStreamInputSource) source).getInputStream(), StandardCharsets.UTF_8);
+                } else {
+                    return null;
+                }
+            }
+
+            return getContentInternal(strinContent);
+        }
+
         protected abstract T getContentInternal(String content) throws Exception;
 
+        /**
+         * @return the instant the cache content of that template was last modified or null if it's unknown
+         * @since 15.8RC1
+         */
```

Comment:
The javadoc should be on the interface no? Not sure why you need it here.

Reply:
Yes. It initially was only exposed here, and I forgot to remove the javadoc when I exposed it also in the interface.